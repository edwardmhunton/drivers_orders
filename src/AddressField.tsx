import React from "react";

export class AddressField extends React.Component {
  constructor(props) {
    super(props);
    this.myRef = React.createRef();
    this.handleBlur = this.handleBlur.bind(this);
  }

  handleBlur() {
    const { changeOrderLocation, order } = this.props;
    changeOrderLocation(order.id, this.myRef.current.value);
  }
  render() {
    const { order } = this.props;
    return (
      <div>
        <p>Pickup Address (free text)</p>
        <form>
          <input
            ref={this.myRef}
            type="text"
            onBlur={this.handleBlur}
            defaultValue={
              order.PickupAddress.coordinates.lat +
              ", " +
              order.PickupAddress.coordinates.lng
            }
          />
        </form>
      </div>
    );
  }
}
