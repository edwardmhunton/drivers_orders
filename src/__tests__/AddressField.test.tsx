import React from "react";
import renderer from "react-test-renderer";

import { AddressField } from "../AddressField";

describe("the fist test", () => {
  it("Renders Correctly", () => {
    const order = {
      id: "12345",
      PickupTime: "1556548991000",
      PickupAddress: {
        coordinates: { lat: "51.5325764", lng: "-0.0578597" },
        street_address: {
          post_code: "EX203LZ",
          address_line_1: "76 Kenworthy Road",
          city: "London"
        }
      },

      PackageSize: {
        length: "1000",
        width: "1000",
        height: "1000"
      }
    };
    const AddressFieldComponent = renderer
      .create(<AddressField order={order} />)
      .toJSON();
    expect(AddressFieldComponent).toMatchSnapshot();
  });
});
