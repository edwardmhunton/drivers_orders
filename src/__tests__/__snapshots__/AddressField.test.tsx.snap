// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the fist test Renders Correctly 1`] = `
<form>
  <input
    defaultValue="51.5325764, -0.0578597"
    onBlur={[Function]}
    type="text"
  />
</form>
`;
